solanaj/src/main/java/org/p2p/solanaj/kits/transaction/parser/ConfirmedTransactionRootParser.kt

```diff
@@ -0,0 +1,201 @@
+
+        return when {
+            orcaSwapInstructionParser.isTransactionContainsOrcaSwap(transactionRoot) -> {
+                listOfNotNull(
+                    orcaSwapInstructionParser.parse(signature, transactionRoot)
+                        .onFailure(onErrorLogger)
+                        .getOrNull()
+                )
+            }
+            serumSwapInstructionParser.isTransactionContainsSerumSwap(transactionRoot) -> {
+                listOfNotNull(
+                    serumSwapInstructionParser.parse(signature, transactionRoot)
+                        .onFailure(onErrorLogger)
+                        .getOrNull()
+                )
+            }
+            else -> {
+                parseTransaction(transactionRoot, signature)
+            }
+        }
+    }
+
+    private fun parseTransaction(
+        transactionRoot: ConfirmedTransactionRootResponse,
+        signature: String,
+    ): List<TransactionDetails> {
+        return transactionRoot.transaction
+            ?.message
+            ?.instructions
+            ?.mapNotNull { parseInstructionByType(signature, transactionRoot, it) }
+            ?.toMutableList()
+            .orEmpty()
+    }
+
+    private fun parseInstructionByType(
+        signature: String,
+        transactionRoot: ConfirmedTransactionRootResponse,
+        parsedInstruction: InstructionResponse
+    ): TransactionDetails? {
+        val parsedInfo = parsedInstruction.parsed
+        return when (parsedInfo?.type) {
+            "burnChecked" -> {
+                parseBurnOrMintTransaction(
+                    signature = signature,
+                    transactionRoot = transactionRoot,
+                    parsedInfo = parsedInfo
+                )
+            }
+            "transfer", "transferChecked" -> {
+                parseTransferTransaction(
+                    signature = signature,
+                    transactionRoot = transactionRoot,
+                    parsedInfo = parsedInfo,
+                )
+            }
+            "closeAccount" -> {
+                parseCloseTransaction(
+                    parsedInstruction = parsedInstruction,
+                    parsedInfo = parsedInfo,
+                    transactionRoot = transactionRoot,
+                    signature = signature,
+                )
+            }
+            "create" -> {
+                parseCreateAccountTransaction(
+                    signature = signature,
+                    transactionRoot = transactionRoot
+                )
+            }
+            else -> {
+                parseUnknownTransaction(
+                    signature = signature,
+                    transactionRoot = transactionRoot
+                )
+            }
+        }
+    }
+
+    private fun parseCreateAccountTransaction(
+        signature: String,
+        transactionRoot: ConfirmedTransactionRootResponse
+    ): CreateAccountDetails {
+        return CreateAccountDetails(
```

Comment:
as you mentioned in my PR you can use here expression body  x D

Reply:
i would never suggest using expression body 🙉